### PR TITLE
clients: move native library building to zig

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: src/clients/dotnet/TigerBeetle/TigerBeetle.csproj [hunk 1]
```diff
@@ -15,23 +15,18 @@
   <PropertyGroup>
     <OS Condition="$([MSBuild]::IsOSPlatform('Windows'))">Windows</OS>
   </PropertyGroup>
```
Comment (batiati) [inline src/clients/dotnet/TigerBeetle/TigerBeetle.csproj]: We can get rid of that too!!

```suggestion
```

File: src/clients/java/pom.xml [hunk 2]
```diff
@@ -233,23 +233,6 @@
...
-            </goals>
-            <configuration>
-              <arguments>
-                <argument>build</argument>
-                <argument>java_client</argument>
-                <argument>-Drelease</argument>
-                <argument>-Dconfig=production</argument>
-              </arguments>
-            </configuration>
-          </execution>
```
Comment (batiati) [inline src/clients/java/pom.xml:252]: Wouldn't it be better to remove this entire `<plugin>` section and run the JNI tests from the CI as well?

Comment (matklad) [reply src/clients/java/pom.xml:252]: Ahh, that `test:jni` did look suspicious to me, but I failed to parse the whole thing as `zig build test:jni` invocation. XML is verbose!

**Review by batiati — DISMISSED** [review]
Nice! The [HACKING.md](https://github.com/tigerbeetle/tigerbeetle/blob/main/docs/HACKING.md#hacking-on-clients) file already points to the CI script, so we don't need to change any instructions for building the clients!

**Comment by batiati** [comment]
>To complicate things, some languages trigger the native build automatically, while others require manually running ./zig/zig build lang_client.

Although it could be changed here ....

**Review by batiati — APPROVED** [review]
> Although it could be changed here ....

Thanks, and sorry for the last-minute change :sweat_smile:!!!

**Comment by matklad** [comment]
No worries, I can fixup&rebase&force-push at the speed of light!

**Comment by matklad** [comment]
(and sometimes I even do that correctly!)

**Pull request merged**